lib/vaahextendflutter/base/base_controller.dart

```diff
@@ -9,9 +9,9 @@ import 'package:sentry_flutter/sentry_flutter.dart';
 import '../app_theme.dart';
 import '../env/env.dart';
 import '../services/api.dart';
-import '../services/dynamic_links.dart';
 import '../services/notification/internal/notification.dart';
 import '../services/notification/push/notification.dart';
+import 'root_assets_controller.dart';
 
 class BaseController extends GetxController {
```

Comment:
We can do initialization something like this, just a suggestion

```dart
class AppConfigs {
  AppConfigs({
    required this.configs,
  });

  final Env env;
}

-------------------------------------------------------------------

abstract class LaunchTask {
  const LaunchTask();

  Future<void> initialize(AppConfigs configs);
  Future<void> dispose();
}

class AppLauncher {
  AppLauncher({
    required this.configs,
  });

  final AppConfigs configs;
  final List<LaunchTask> tasks = [];

  void addTask(LaunchTask task) {
    tasks.add(task);
  }

  void addTasks(Iterable<LaunchTask> tasks) {
    tasks.addAll(tasks);
  }

  Future<void> launch() async {
    for (final task in tasks) {
      await task.initialize(configs);
    }
  }

  Future<void> dispose() async {
    for (final task in tasks) {
      await task.dispose();
    }
    tasks.clear();
  }
}

-------------------------------------------------------------------

class BaseController extends GetxController{

  void init(AppConfigs configs)
    final launcher = AppLauncher(configs: configs);
    launcher.addTasks(
     const AlgoliTask();
     .
     .
     .
     // other services
    );
  }
}


-------------------------------------------------------------------

class AlgoliTask extends LaunchTask {
  const AlgoliTask();

  @override
  Future<void> initialize(AppConfigs configs) async {
  // initialize algolia service and add the object into di or singleton class
  // this way other services can utilize this object from  di or singleton class
  }

  @override
  Future<void> dispose() async {}
}
```

Reply:
Good proposal, can we have a separate PR for this one or can you create a separate commit on my branch?

Reply:
Also will we have to do some testing around this?

Reply:
sure

Reply:
Let's go with this approach for now. I'll create new PR for my suggestion.